mantidproject/mantid
# Use RunWidget in Data Processor interface

### Description of work
This PR adds the runwidget as a substitute for the run button on the tabs of the Data Processor interface. The presenter of the runwidget is held by the parent data processor class. The view of the run button is created on the automatic ui file created by qt. 
Additionally, there's a refactor of the 




Part of #37381. 




#### Further detail of work


### To test:
- Open Data Processor interface. 
- Click on every `Run` button of every tab withouth adding any data to the bats. Every click should produce a floating
  warning message telling you why you can't run the algorithm.
- Follow `DataProcessor` manual test: https://developer.mantidproject.org/Testing/Inelastic/DataProcessorTests.html







---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
